proxy.c

```diff
@@ -228,20 +228,23 @@ int create_socket(int port) {
     }
 
     if ((server_sock = socket(res->ai_family, res->ai_socktype, res->ai_protocol)) < 0) {
+        freeaddrinfo(res); // Free memory on failure
         return SERVER_SOCKET_ERROR;
     }
 
-
     if (setsockopt(server_sock, SOL_SOCKET, SO_REUSEADDR, &optval, sizeof(optval)) < 0) {
+        freeaddrinfo(res);
         return SERVER_SETSOCKOPT_ERROR;
     }
 
     if (bind(server_sock, res->ai_addr, res->ai_addrlen) == -1) {
-            close(server_sock);
+        close(server_sock);
```

Comment:
@gaurovgiri I think `freeaddrinfo(res)` should also be used here after `close(server_sock)`, right? Please correct me if I'm missing something.

Reply:
oh yes, you are right. I will fix it

Reply:
I've updated the code